File: frontend/src/types/match.ts
export type MatchParticipant = {
  id: number;
  friend_id: number | null;
  display_name: string;
  game_name: string;
  tag_line: string;
  champion_id: number | null;
  champion_name: string | null;
  team_id: number;
  kills: number;
  deaths: number;
  assists: number;
  damage_to_champions: number;
  win: boolean;
  is_baboon: boolean;
};

export type MatchSummary = {
  id: number;
  riot_match_id: string;
  queue_id: number;
  game_end_time: string;
  duration_seconds: number;
  game_version: string | null;
  registered_friend_count: number;
  team_won: boolean | null;
  lowest_damage_to_champions: number | null;
  participants: MatchParticipant[];
  baboons: MatchParticipant[];
};

export type PaginatedMatches = {
  items: MatchSummary[];
  limit: number;
  offset: number;
  total: number;
};

export type MatchDetail = {
  id: number;
  riot_match_id: string;
  queue_id: number;
  game_creation: string | null;
  game_start_time: string | null;
  game_end_time: string;
  duration_seconds: number;
  game_version: string | null;
  imported_at: string;
  participants: MatchParticipant[];
  baboons: MatchParticipant[];
};

export type CurrentBaboonPlayer = {
  friend_id: number | null;
  display_name: string;
  game_name: string;
  tag_line: string;
  champion_id: number | null;
  champion_name: string | null;
  kills: number;
  deaths: number;
  assists: number;
  damage_to_champions: number;
  win: boolean;
};

export type CurrentBaboonResponse = {
  has_current_baboon: boolean;
  match: {
    id: number;
    riot_match_id: string;
    game_end_time: string;
    duration_seconds: number;
  } | null;
  baboons: CurrentBaboonPlayer[];
};

export type MatchSyncSummary = {
  status: string;
  friends_checked: number;
  candidate_match_ids: number;
  new_candidates_examined: number;
  matches_imported: number;
  matches_already_known: number;
  matches_skipped: number;
  skipped_reasons: Record<string, number>;
  imported_match_ids: string[];
  current_baboons: Array<{
    display_name: string;
    damage_to_champions: number;
  }>;
};

File: backend/app/schemas/match.py
from datetime import datetime

from pydantic import BaseModel


class MatchParticipantRead(BaseModel):
    id: int
    friend_id: int | None
    display_name: str
    game_name: str
    tag_line: str
    champion_id: int | None
    champion_name: str | None
    team_id: int
    kills: int
    deaths: int
    assists: int
    damage_to_champions: int
    win: bool
    is_baboon: bool


class MatchSummary(BaseModel):
    id: int
    riot_match_id: str
    queue_id: int
    game_end_time: datetime
    duration_seconds: int
    game_version: str | None
    registered_friend_count: int
    team_won: bool | None
    lowest_damage_to_champions: int | None
    participants: list[MatchParticipantRead]
    baboons: list[MatchParticipantRead]


class MatchListResponse(BaseModel):
    items: list[MatchSummary]
    limit: int
    offset: int
    total: int


class MatchDetail(BaseModel):
    id: int
    riot_match_id: str
    queue_id: int
    game_creation: datetime | None
    game_start_time: datetime | None
    game_end_time: datetime
    duration_seconds: int
    game_version: str | None
    imported_at: datetime
    participants: list[MatchParticipantRead]
    baboons: list[MatchParticipantRead]


class CurrentBaboonMatch(BaseModel):
    id: int
    riot_match_id: str
    game_end_time: datetime
    duration_seconds: int


class CurrentBaboonPlayer(BaseModel):
    friend_id: int | None
    display_name: str
    game_name: str
    tag_line: str
    champion_id: int | None
    champion_name: str | None
    kills: int
    deaths: int
    assists: int
    damage_to_champions: int
    win: bool


class CurrentBaboonResponse(BaseModel):
    has_current_baboon: bool
    match: CurrentBaboonMatch | None
    baboons: list[CurrentBaboonPlayer]


class SyncBaboonSummary(BaseModel):
    display_name: str
    damage_to_champions: int


class MatchSyncSummary(BaseModel):
    status: str
    friends_checked: int
    candidate_match_ids: int
    new_candidates_examined: int
    matches_imported: int
    matches_already_known: int
    matches_skipped: int
    skipped_reasons: dict[str, int]
    imported_match_ids: list[str]
    current_baboons: list[SyncBaboonSummary]
